packages/myst-cli/src/session/session.ts

```diff
@@ -157,4 +164,33 @@ export class Session implements ISession {
     });
     return warnings;
   }
+
+  async jupyterSessionManager(): Promise<SessionManager | undefined> {
+    if (this._jupyterSessionManager !== null) {
+      return Promise.resolve(this._jupyterSessionManager);
+    }
+    try {
+      const partialServerSettings = await new Promise<JupyterServerSettings>(
+        async (resolve, reject) => {
+          if (process.env.JUPYTER_BASE_URL === undefined) {
+            resolve(findExistingJupyterServer() || (await launchJupyterServer(this.contentPath(), this.log)));
+          } else {
+            resolve({
+              baseUrl: process.env.JUPYTER_BASE_URL,
+              token: process.env.JUPYTER_TOKEN,
+            });
+          }
+        },
+      );
+      const serverSettings = ServerConnection.makeSettings(partialServerSettings);
+      const kernelManager = new KernelManager({ serverSettings });
+      const manager = new SessionManager({ kernelManager, serverSettings });
+      // TODO: this is a race condition, even though we shouldn't hit if if this promise is actually awaited
+      this._jupyterSessionManager = manager;
+      return manager;
+    } catch {
+      this._jupyterSessionManager = undefined;
+      return undefined;
+    }
+  }
```

Comment:
This does not feel elegant to me, but I want to get "something" in place for us to talk about.